src/program.rs

```diff
@@ -350,6 +420,41 @@ macro_rules! declare_builtin_function {
     };
 }
 
+/// Tombstone function for when we call an invalid syscall (e.g. a syscall deactivated through a
+/// feature gate
+#[derive(Clone, PartialEq)]
+pub struct SyscallTombstone {}
+impl SyscallTombstone {
+    /// Tombstone function entrypoint
+    #[allow(clippy::arithmetic_side_effects)]
+    pub fn vm<T: ContextObject>(
+        ctx: *mut EbpfVm<T>,
+        _arg1: u64,
+        _arg2: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+    ) {
+        let vm = unsafe {
+            &mut *(ctx
+                .cast::<u64>()
+                .offset(-(get_runtime_environment_key() as isize))
+                .cast::<EbpfVm<T>>())
+        };
+        let config = vm.loader.get_config();
+        if config.enable_instruction_meter {
+            vm.context_object_pointer
+                .consume(vm.previous_instruction_meter - vm.due_insn_count);
+        }
+        vm.program_result = ProgramResult::Err(EbpfError::SyscallError(Box::new(
```

Comment:
could also add a `debug_assert!(false);` to indicate that this is never supposed to actually run.